src/decompress.rs

```diff
@@ -188,6 +247,79 @@ pub fn decompress(jpeg_data: &[u8], format: PixelFormat) -> Result<Image<Vec<u8>
     Ok(image)
 }
 
+/// Decompress a JPEG image to YUV.
+///
+/// Returns a newly allocated yuv image
+///
+/// # Example
+///
+/// ```
+/// // read JPEG data from file
+/// let jpeg_data = std::fs::read("examples/parrots.jpg")?;
+///
+/// // decompress the JPEG into YUV image
+/// let image = turbojpeg::decompress_to_yuv(&jpeg_data)?;
+/// assert_eq!((image.width, image.height), (384, 256));
+/// assert_eq!(image.pixels.len(), 294912);
+///
+/// # Ok::<(), Box<dyn std::error::Error>>(())
+/// ```
+pub fn decompress_to_yuv(jpeg_data: &[u8]) -> Result<YUVImage<Vec<u8>>> {
+    let mut decompressor = Decompressor::new()?;
+    let header = decompressor.read_header(jpeg_data)?;
+    let yuv_video_pad = 4;
+    let yuv_frame_size = match yuv_frame_size(
+        header.width,
+        yuv_video_pad,
+        header.height,
+        header.subsamp,
+    ) {
+        Ok(size) => size,
+        Err(e) => panic!("{}", e)
+    };
+
+    let mut yuv_image = YUVImage {
+        pixels: vec![0; yuv_frame_size],
+        width: header.width,
+        pad: yuv_video_pad,
+        height: header.height,
+    };
+    decompressor.decompress_to_yuv(jpeg_data, yuv_image.as_deref_mut())?;
+
+    Ok(yuv_image)
+}
+
+/// Determine size in bytes of a yuv image
+///
+/// Returns size in bytes of a yuv image
+/// 
+/// # Example
+///
+/// ```
+/// // read JPEG data from file
+/// let jpeg_data = std::fs::read("examples/parrots.jpg")?;
+///
+/// // read the JPEG header
+/// let header = turbojpeg::read_header(&jpeg_data)?;
+/// // get yuv frame size
+/// let yuv_video_pad = 4;
+/// let yuv_frame_size = turbojpeg::yuv_frame_size(header.width, yuv_video_pad, header.height, header.subsamp);
+/// assert_eq!(yuv_frame_size.unwrap(), 294912);
+///
+/// # Ok::<(), Box<dyn std::error::Error>>(())
+/// ```
+pub fn yuv_frame_size(width: usize, pad: usize, height: usize, subsamp: Subsamp) -> Result<usize> {
```

Comment:
Perhaps a more consistent name for this function would be `yuv_pixels_len()`, because it returns the expected length of `YuvImage::pixels`? (In the same way that `tjBufSize()` is exposed as `compressed_buf_len()`)